### PR TITLE
Add cascadelake implementation, which use AVX512 Intrinsics to optimi…

### PR DESCRIPTION
**Pull request by mellonyou**
Add cascadelake implementation, which use AVX512 Intrinsics to optimization the performance.

In order to make full use of the performance of avx512, we use the avx512 intrinsic interface to rewrite the “Simdjson” code, and the related modifications are classified into the “Cascadelake” implementation. 
After testing, the performance of “Cascadelake” implementation is about 10% better than that of “Haswell” implementation.

### CODE REVIEW

**Comment by lemire** [comment]
At a glance, this looks very good !!!

Could you have a look at 

doc/performance.md

At the end, it says...

> The simdjson library does not currently support AVX-512 instructions and it does not make use of heavy 256-bit instructions. We do use vectorized multiplications, but only using 128-bit registers. Thus there should be no downclocking due to simdjson on recent processors.

I think we can replace it with...

> The simdjson library does not make use of heavy 256-bit instructions. We do use vectorized multiplications, but only using 128-bit registers. Thus there should be no downclocking due to simdjson on recent processors, except when AVX-512 is detected.

**Comment by lemire** [comment]
@mellonyou Could you please see about updating `performance.md` (the second last paragraph) so that it remains truthful.


There are various optimisations that could be added. For example, this code won't use `vpternlogq` and so forth, but this can be added later.

File: include/simdjson/cascadelake/simd.h [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+    // Copies to 'output" all bytes corresponding to a 0 in the mask (interpreted as a bitset).
+    // Passing a 0 value for mask would be equivalent to writing out every byte to output.
+    // Only the first 32 - count_ones(mask) bytes of the result are significant but 32 bytes
+    // get written.
+    // Design consideration: it seems like a function with the
+    // signature simd8<L> compress(uint32_t mask) would be
+    // sensible, but the AVX ISA makes this kind of approach difficult.
+    template<typename L>
+    simdjson_really_inline void compress(uint64_t mask, L * output) const {
```
Comment (lemire) [inline include/simdjson/cascadelake/simd.h:124]: Sadly cascade lake won't support vpcompressb. 

(Not a request for change, but you could add a comment to that effect.)

**Comment by lemire** [comment]
Once this is merged, we will be able to further optimize, see https://github.com/simdjson/simdjson/issues/1812

File: include/simdjson/cascadelake/simd.h (hunk 1 above)
Comment (mellonyou) [reply include/simdjson/cascadelake/simd.h:126]: Have added a comment, and I will do some investigation on vpcompressb later.

**Comment by mellonyou** [comment]
> to
I have updated the performance.md. And I will do some investigation on vpternlogq later.

**Comment by mellonyou** [comment]
> @mellonyou Could you please see about updating `performance.md` (the second last paragraph) so that it remains truthful.
> 
> There are various optimisations that could be added. For example, this code won't use `vpternlogq` and so forth, but this can be added later.

I have updated the performance.md. And I will do some investigation on vpternlogq later.

**Comment by lemire** [comment]
Ok. I will be merging this to a local branch for further investigation before merging in the main branch.

**Pull request merged**